### DISCUSSION TITLE
Validating empty strings and lists to `None`

### DISCUSSION THREAD

**Discussion by mirandrom** [pydantic/pydantic#2611] [Question] [answered]
Hello, I'm using pydantic in an open-source project ([PyS2](https://github.com/mirandrom/pys2)) that wraps the Semantic Scholar API, and it's been great so thank you for the wonderful library. The issue I'm running into is that the API sometimes returns empty types (e.g. "" for strings, or [] for lists) and sometimes returns `None` for the same fields. Is there a way to define validators on the type-level (e.g. for all strings instead of for each individual string-type field) so that empty types consistently resolve to `None` or vice-versa?

**Comment by mirandrom** [comment, marked as answer]
This kind of worked for me. 

I'd still be curious if there was a more general solution that:
- is only called for fields of a specific type;
- allows me to convert ``None`` to the default value of that field's type.

```python
from pydantic import BaseModel, validator
from typing import Iterable


class NoneModel(BaseModel):
    @validator('*', pre=True)
    def convert_to_none(cls, v):
        if isinstance(v, str) and v.strip() == "":
            return None
        if isinstance(v, Iterable) and len(v) == 0:
            return None
        else:
            return v
```

**Reply by PrettyWood** [reply]
I would have done something similar. For now there is no easier solution AFAIK
